src/ledger-bridge-keyring.ts

```diff
@@ -0,0 +1,883 @@
+  }
+
+  async #getAccountsBIP44(from: number, to: number) {
+    const accounts: {
+      address: string;
+      balance: number | null;
+      index: number;
+    }[] = [];
+
+    for (let i = from; i < to; i++) {
+      const path = this.#getPathForIndex(i);
+      const address = await this.unlock(path);
+      const valid = this.implementFullBIP44
+        ? await this.#hasPreviousTransactions(address)
+        : true;
+      accounts.push({
+        address,
+        balance: null,
+        index: i,
+      });
+
+      // PER BIP44
+      // "Software should prevent a creation of an account if
+      // a previous account does not have a transaction history
+      // (meaning none of its addresses have been used before)."
+      if (!valid) {
+        break;
+      }
+    }
+    return accounts;
+  }
+
+  #getAccountsLegacy(from: number, to: number) {
+    const accounts: {
+      address: string;
+      balance: number | null;
+      index: number;
+    }[] = [];
+
+    for (let i = from; i < to; i++) {
+      const address = this.#addressFromIndex(pathBase, i);
+      accounts.push({
+        address,
+        balance: null,
+        index: i,
+      });
+      this.paths[ethUtil.toChecksumAddress(address)] = i;
+    }
+    return accounts;
+  }
+
+  // eslint-disable-next-line no-shadow
+  #addressFromIndex(basePath: string, i: number) {
+    const dkey = this.hdk.derive(`${basePath}/${i}`);
+    const address = ethUtil
+      .publicToAddress(dkey.publicKey, true)
+      .toString('hex');
+    return ethUtil.toChecksumAddress(`0x${address}`);
+  }
+
+  #pathFromAddress(address: string) {
+    const checksummedAddress = ethUtil.toChecksumAddress(address);
+    let index = this.paths[checksummedAddress];
+    if (typeof index === 'undefined') {
+      for (let i = 0; i < MAX_INDEX; i++) {
+        if (checksummedAddress === this.#addressFromIndex(pathBase, i)) {
+          index = i;
+          break;
+        }
+      }
+    }
+
+    if (typeof index === 'undefined') {
+      throw new Error('Unknown address');
+    }
+    return this.#getPathForIndex(index);
+  }
+
+  #getPathForIndex(index: number) {
+    // Check if the path is BIP 44 (Ledger Live)
+    return this.#isLedgerLiveHdPath()
+      ? `m/44'/60'/${index}'/0/0`
+      : `${this.hdPath}/${index}`;
+  }
+
+  #isLedgerLiveHdPath() {
+    return this.hdPath === `m/44'/60'/0'/0/0`;
+  }
+
+  #toLedgerPath(path: string) {
+    return path.toString().replace('m/', '');
+  }
+
+  async #hasPreviousTransactions(address: string) {
+    const apiUrl = this.#getApiUrl();
+    const response = await window.fetch(
+      `${apiUrl}/api?module=account&action=txlist&address=${address}&tag=latest&page=1&offset=1`,
+    );
+    const parsedResponse = await response.json();
+    if (parsedResponse.status !== '0' && parsedResponse.result.length > 0) {
+      return true;
+    }
+    return false;
+  }
+
+  #getApiUrl() {
+    return NETWORK_API_URLS[this.network] || NETWORK_API_URLS.mainnet;
```

Comment:
Nit: It should not be possible for `NETWORK_API_URLS[this.network]` to be undefined if  `this.network` is typed correctly, so you may not need the `|| NETWORK_API_URLS.mainnet`. If TypeScript doesn't complain then it's fine to leave this in too. Up to you though.